crates/ruff_python_semantic/src/red_knot/semantic_index/symbol.rs

```diff
@@ -84,62 +76,70 @@ bitflags! {
 }
 
 /// ID that uniquely identifies a public symbol defined in a module's root scope.
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+#[salsa::tracked]
 pub struct PublicSymbolId {
-    scope: ScopeId,
-    symbol: ScopeSymbolId,
-}
-
-impl PublicSymbolId {
-    pub(crate) fn new(scope: ScopeId, symbol: ScopeSymbolId) -> Self {
-        Self { scope, symbol }
-    }
-
-    pub fn scope(self) -> ScopeId {
-        self.scope
-    }
-
-    pub(crate) fn scope_symbol(self) -> ScopeSymbolId {
-        self.symbol
-    }
-}
-
-impl From<PublicSymbolId> for ScopeSymbolId {
-    fn from(val: PublicSymbolId) -> Self {
-        val.scope_symbol()
-    }
+    pub(crate) file: VfsFile,
+    pub(crate) scope_id: ScopeSymbolId,
```

Comment:
I renamed them like 5 times and all of them are awkward, either when reading it at the definition or add the usage side. 

For example, I often write

```rust
let symbol = public_symbol_ty(...);
symbol.symbol_id()
```

The `symbol_id` is then just confusing because why do I get the symbol id when I already have one? 


>  (Maybe ScopeSymbolId should rather be ScopedSymbolId?)

Yeah maybe. But should it then be `FiledSymbolID` :P I think renaming to scoped makes sense regardless of the slight inconsistency but I'll do it as a separate PR